### DISCUSSION TITLE
Can not set both UCP_OP_ATTR_FIELD_DATATYPE and UCP_OP_ATTR_FLAG_NO_IMM_CMPL to ucp_am_send_nbx()?

### DISCUSSION THREAD

**Discussion by huzhijiang** [openucx/ucx#9791] [Q&A]
If set both UCP_OP_ATTR_FIELD_DATATYPE and UCP_OP_ATTR_FLAG_NO_IMM_CMPL flags, the following line of code will always treat data type as contig? iov data type will be ignored ?

https://github.com/openucx/ucx/blob/d4bb985e310b88c9eb767e539adfd3237bed760e/src/ucp/core/ucp_am.c#L997

**Comment by yosefe** [comment]
Seems like a bug; would need to apply same logic as in https://github.com/openucx/ucx/blob/d4bb985e310b88c9eb767e539adfd3237bed760e/src/ucp/tag/tag_send.c#L273-L280
